rust/core-lib/src/editor.rs

```diff
@@ -357,8 +359,174 @@ impl Editor {
                     && config.use_tab_stops {
                     tab_start
                 } else {
-                    self.text.prev_grapheme_offset(region.end)
-                        .unwrap_or(region.end)
+
```

Comment:
This fn is definitely too long now. I might consider moving this to `movement.rs`, or I might even consider adding a `backspace.rs` file?

Reply:
That's from Android's [getOffsetForBackspaceKey:249](https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/text/method/BaseKeyListener.java#249). I need to write a function to get combining class (maybe in future work?) . However, without this, I passed all tests provided by Android :laughing: .

Reply:
I don't think `movement.rs`  is a good place for it. Should I move the whole `delete_backward` function to a new file or just some parts of it? 

If break up it into multiple functions, I cannot even think up proper function names of them :cry: .

Reply:
So yes, I could imagine moving the whole function elsewhere, or at least everything that replaces `prev_grapheme_offset`. You might call this, `fn offset_for_delete_backwards(offset: usize, text: &Rope) -> usize`.

I think this probably belongs in core, somewhere. A new file would be fine, `backspace.rs`?

`State` gets a `fn execute()` or `fn run()` or fn

Reply:
OK! I have moved it.